### DISCUSSION TITLE
Zotero integration

### DISCUSSION THREAD

**Discussion by retorquere** [asreview/asreview#735] [Ideas]
@ottomattas wrote:

> Thanks for bringing your wish out @ ckolar. **The feature is also being implemented with Zotero in mind**. Hopefully, we can give you good news rather sooner than later.

This sounded to me like work is already underway to build some kind of integration with Zotero, and I don't want to duplicate efforts.

> I am a strong proponent for the idea of automating everything while I also see a big challenge here. How do we determine which software to develop for and support going forward? With import/export as a transport mechanism, we leave more freedom of choice to the user while the feature set might be less complete. That said, I'd be very interested in collaborating on some automation between open sourced software tools like Zotero and ASReview.
>
> I invite you to create a new issue for this so we can start the specific discussion at the very least! 🚀

I would say that as long as there is a stable API that can do what the download and upload can do, it is up to the plugin developer (me in this case) to make use of it. I could build a plugin that would allow a Zotero user to right-click on a collection, select "send to ASReview" (or something to that effect), which would generate a CSV or RIS and post it to an API, replacing the save-and-upload. It all depends on how much pre-upload activity the user would be expected to do, usually -- if the user must first select a project, then some settings, and only then upload, an integration of this kind would need to replicate all that UI activity, and that seems like a waste.

In any case, the first work I'll be doing on a Zotero plugin would be to read back the results from ASReview and mark the items in Zotero with it so it can be used for sorting in Zotero. I'm working on this with [PROSEC](https://prosec.network/?lang=en).

**Comment by ottomattas** [comment]
> I would say that as long as there is a stable API that can do what the download and upload can do, it is up to the plugin developer (me in this case) to make use of it. 
> ...
> In any case, the first work I'll be doing on a Zotero plugin would be to read back the results from ASReview and mark the items in Zotero with it so it can be used for sorting in Zotero. I'm working on this with [PROSEC](https://prosec.network/?lang=en).

This sounds like a cool effort not only from Zotero side but also ASReview. Developing extensions is a great way not only to solve a problem but to create open discussions around the idea. That said, the API is going to change in the upcoming release [a little](https://github.com/asreview/asreview/pull/682/files). The change will have to do with the RIS exporting feature currently being implemented which needs to be added to the API.

This should not have consequences for the sorting idea you proposed. Something that might be relevant, though - we are also addressing a deeper challenge around internally labelling our records in ASReview.

Are you already getting started and maybe want look at our API together before coding your plugin?

**Reply by retorquere** [reply]
I'm about to get started, but a part of the work is independent from the api, as I need to figure out a way to cleanly note the review ranking in the zotero items.

**Reply by ottomattas** [reply]
Awesome. Let us know when you are starting to look at our API and want to discuss something. Always happy to collaborate.
